src/UploadImage.tsx

```diff
@@ -11,27 +11,32 @@ interface Props {
   ) => void;
 }
 
-export default class Upload3DImage extends Component<Props> {
+export default class UploadImage extends Component<Props> {
   private imageFileInfo: ImageFileInfo | null;
 
-  private slicesData: Array<Array<ImageBitmap>>;
+  private slicesData: Array<Array<ImageBitmap>> = [];
 
   constructor(props: Props) {
     super(props);
     this.imageFileInfo = null;
   }
 
   private uploadImage = (imageFile: File): void => {
-    this.readFile(imageFile)
-      .then((buffer: ArrayBuffer) => {
-        this.imageFileInfo = new ImageFileInfo(imageFile.name);
-        this.loadImageFile(buffer).catch((error) => {
+    const patternTiff = /tiff|tif$/i;
+    const patternNonTiff = /jpg|jpeg|png|gif$/i;
+    this.imageFileInfo = new ImageFileInfo(imageFile.name);
+
+    if (patternTiff.exec(imageFile.name)) {
+      this.readFile(imageFile)
+        .then((buffer: ArrayBuffer) => {
+          this.loadTiffImage(buffer);
+        })
+        .catch((error) => {
           console.log(error);
         });
-      })
-      .catch((error) => {
-        console.log(error);
-      });
+    } else if (patternNonTiff.exec(imageFile.name)) {
```

Comment:
Rather than else if here and checking against a pattern, could just try to load the image anyway...this will let us handle formats that the browser supports, but aren't in that regex (eg webp)?

Reply:
Done